XenAdmin/SettingsPanels/GpuEditPage.cs

```diff
@@ -338,8 +345,8 @@ public VGpuDetailRow(VGPU vGpu)
 
         private void SetCells()
         {
-            if (int.TryParse(VGpu.device, out var device) && device != 0)
-                deviceColumn.Value = device;
+            if (int.TryParse(VGpu.device, out var device))
+                deviceColumn.Value = Helpers.PlymouthOrGreater(VGpu.Connection) ? device + VGPU_PCI_SLOT_NUMBER_SHIFT : device;

```

Comment:
What happens if the host is earlier than plymouth and the device is 0? In that case 0 means unassigned, shouldn't we use the hyphen?

Reply:
In that case, `0` actually means `11`, which is hard coded.
But per my test, when XC connect to a older XS with only 1 vgpu, the device id is never shown at all.